src/main/java/it/uniroma3/mat/extendedset/intset/ConciseSetUtils.java

```diff
@@ -445,12 +445,17 @@ private void resetZeroSequence(int offset, int word, boolean fromBeginning)
 
     private void resetLiteral(int offset, int word, boolean fromBeginning)
     {
-      len = 0;
-      for (int i = 0; i < MAX_LITERAL_LENGTH; i++) {
-        if ((word & (1 << i)) != 0) {
-          buffer[len++] = offset + i;
-        }
+      int len = 0;
+      word &= ALL_ONES_WITHOUT_MSB;
+      int[] buffer = this.buffer;
+      while (word != 0) {
+        int trailingZeros = Integer.numberOfTrailingZeros(word);
+        offset += trailingZeros;
+        buffer[len++] = offset;
+        word >>>= trailingZeros;
```

Comment:
Since we're shaving operations, you've already cleared the high bit here, do you really need unsigned shift?